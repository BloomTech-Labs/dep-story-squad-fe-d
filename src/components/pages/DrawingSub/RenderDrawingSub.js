import React from 'react';
import { Header } from '../../common';
import { Row } from 'antd';
import { UploadDocs } from '../../common/';
import { postNewDrawingSub } from '../../../api/index';
import { SubmissionModal } from '../../common/index';

const RenderDrawingSub = () => {
  const inst =
    'Once you finish your drawing, please take a picture of all of your pages and upload them. After all pages are uploaded, click submit.';
  const submissionId = 1;

  return (
    <>
      <Header title="READY, SET ...DRAW!" />
      <SubmissionModal instructions={inst} />
      <div className="writing-sub-container">
        <Row className="main-row">
          <p>Draw a picture of your favorite part of the story.</p>
        </Row>
        <div className="upload">
          <h1>Upload</h1>
          <UploadDocs
            submitButtonClassname="orange-submit-button"
            uploadButtonText="Choose files from your device"
            uploadButtonClassname="uploadButton"
            fileName="drawing"
            apiAxios={postNewDrawingSub}
            submissionId={submissionId}
          />
        </div>
      </div>
    </>
  );
};

export default RenderDrawingSub;
