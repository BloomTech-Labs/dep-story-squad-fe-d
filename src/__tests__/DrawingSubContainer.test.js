import * as React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import { cleanup } from '@testing-library/react';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import LoadingComponent from '../components/common/ParentLoadingComponent';
import RenderDrawingSub from '../components/pages/DrawingSub/RenderDrawingSub';
import DrawingSubContainer from '../components/pages/DrawingSub/DrawingSubContainer';

afterEach(() => {
  cleanup();
});

jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true,
      },
      authService: {},
    };
  },
}));

configure({ adapter: new Adapter() });

describe('<ProfileModalContainer />', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore();

  describe('Render <ProfileModalContainer />', () => {
    let shallowWrapper;
    beforeEach(() => {
      shallowWrapper = shallow(
        <Router>
          <Provider store={store}>
            <DrawingSubContainer />
          </Provider>
        </Router>
      ).dive();
    });

    it('Find RenderDrawingSub', () => {
      expect(shallowWrapper.find(RenderDrawingSub));
    });
    it('Find Loading Component', () => {
      expect(shallowWrapper.find(LoadingComponent));
    });
    it('Find DrawingSubContainer', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
