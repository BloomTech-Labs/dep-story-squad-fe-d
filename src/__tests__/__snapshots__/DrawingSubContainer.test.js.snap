// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileModalContainer /> Render <ProfileModalContainer /> Find DrawingSubContainer 1`] = `ShallowWrapper {}`;

exports[`Render <DrawingSubContainer /> Find ParentDashboard 1`] = `ShallowWrapper {}`;

exports[`Render <DrawingSubContainer /> Find drawingSubContainer 1`] = `ShallowWrapper {}`;
